src/main/haskell/language-kore/src/Data/Kore/Implicit/ImplicitSorts.hs

```diff
@@ -66,3 +67,26 @@ import           Data.Kore.Implicit.ImplicitSortsImpl
     defineMetaSort PatternSort
 
 stringMetaSort = charListMetaSort
+
+getPatternResultSort :: Pattern level Variable child -> Sort level
```

Comment:
How about moving this to ASTHelpers? Besides this function not being about implicit sorts, implementing ApplicationPattern may require functionality from that file.